sdk/harambe/core.py

```diff
@@ -466,7 +467,14 @@ async def run(
                 await setup(sdk)
 
             if not harness_options.get("disable_go_to_url", False):
-                await page.goto(url)
+                maybe_response = await page.goto(url)
+                if (
+                    maybe_response
+                    and (status := cast(Response, maybe_response).status) >= 400
+                ):
+                    raise RuntimeError(
+                        f"Got an unexpected status code of {status} when attempting to load the page"
+                    )
```

Comment:
I dont think you want to be throwing a runtime error here. This is for internal stuff. Throw a harambeError instead or an HTTP exception

Reply:
Updated to a GotoError which extends harambe error